### PR TITLE
fix(GODT-1729): Fix header parsing of fields without space

### PR DESCRIPTION
**Pull request by LBeernaertProton**
Fix the header parsing of fields that do not have a space after the
colon.

### CODE REVIEW

File: tests/fetch_body_test.go [hunk 1]
```diff
@@ -657,7 +657,7 @@ Errors-to: dovecot-bounce@procontrol.fi
 X-original-sender: tss@iki.fi
 Precedence: bulk
 X-list: dovecot
-X-UID: 3                                                  
+X-UID: 3
```
Comment (cuthix) [inline tests/fetch_body_test.go:660]: why remove whitespace in the end of the line? I think it's part of the test isn't it?

File: tests/fetch_body_test.go [hunk 2]
```diff
@@ -706,6 +706,76 @@ func TestFetchBodyPeekInvalidSectionNumber(t *testing.T) {
...
+Return-Path: <tss@iki.fi>
+X-Original-To: dovecot@procontrol.fi
+Delivered-To: dovecot@procontrol.fi
+Received: from oma.irssi.org (ip213-185-36-189.laajakaista.mtv3.fi [213.185.36.189])
+	by danu.procontrol.fi (Postfix) with ESMTP id 2A7282387F
+	for <dovecot@procontrol.fi>; Mon,  7 Apr 2003 01:27:38 +0300 (EEST)
+Received: by oma.irssi.org (Postfix, from userid 1000)
+	id 3D1565E01F95; Mon,  7 Apr 2003 01:27:36 +0300 (EEST)
+Subject: [dovecot] Re: message order reversed on copying
+From: Timo Sirainen <tss@iki.fi>
```
Comment (cuthix) [inline tests/fetch_body_test.go:723]: I am not 100% sure but this should be also valid header. 

```
From: "Timo =?UTF-8?B?VGhpcyBpcyBmdW5ueSBzdG9yeSBidXQgaXQgaXMgcG9zc2libGUgdG8gcHV0IHF1aXRlIGEgbG9u
ZyB0ZXh0IGludG8gdGhlIHN1YmplY3QgZmllbGQuLiBzbyBsb25nIHRoYXQgYWN0dWFsbHkgbmVl
ZGVkIHRvIGxpbmUgYnJlYWsgaW5zaWRlIHRoZSBxdW90YXRpb25zLg==?= Sirainen" <tss@iki.fi>
```

The idea is that when quotation is opened you don't need to put `\t` in new lines. I know it is used for long attachment names in MIME header. If it is applicable for `From:` field I am not sure.

File: rfc822/header.go [hunk 3]
```diff
@@ -270,15 +270,23 @@ func mergeMultiline(line []byte) string {
 	var res [][]byte
 
 	for line := range forLines(bufio.NewReader(bytes.NewReader(line))) {
-		res = append(res, bytes.TrimSpace(line))
+		trimmed := bytes.TrimSpace(bytes.Trim(line, "\t"))
```
Comment (cuthix) [inline rfc822/header.go:273]: isn't TrimSpace also trimming `\t` ?

File: tests/fetch_body_test.go (hunk 1 above)
Comment (LBeernaertProton) [reply tests/fetch_body_test.go:660]: Whitspace trimming madness. Will revert.

File: rfc822/header.go (hunk 3 above)
Comment (LBeernaertProton) [reply rfc822/header.go:273]: Looks like you are correct. The change must have masked some  other issue I was having at the time. Good catch.

File: tests/fetch_body_test.go (hunk 1 above)
Comment (LBeernaertProton) [reply tests/fetch_body_test.go]: Fixed.

**Review by cuthix — APPROVED** [review]

**Pull request merged**